### DISCUSSION TITLE
Snippets for Copilot Chat

### DISCUSSION THREAD

**Discussion by devcatalin** [community/community#106462] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

*Overview:*
I am proposing a new feature for the GitHub Copilot Chat extension in VSCode that allows users to create custom "snippets."  
They would act as shorthand commands to modify or guide the behavior of Copilot's prompts during a coding session.

*Snippets*
The value assigned to each snippet is essentially a prompt instruction. When a user types this snippet keyword in their query, the Copilot interprets it and applies the corresponding instruction to modify its prompt.

Assuming the snippet keyword would be referenced using the exclamation mark (but it could be any other symbol) - I created some examples below.

*Examples of Custom Snippets:*
1. !concise:
   - Definition: `!concise = "Answer succinctly. One paragraph."`
   - Usage: Typing `!concise Explain closures in JavaScript` - prompts a brief explanation.

2. !code:
   - Definition: `!code = "Provide a code example without additional explanation or comments."`
   - Usage: ` !code Show a Python function for Fibonacci series` - yields only the code.

3. !detailed:
   - Definition: `!detailed = "Provide a comprehensive explanation with details and examples."`
   - Usage: `!detailed How does React's useState hook work?` - results in an in-depth explanation with examples.

4. !debug:
   - Definition: `!debug = "Suggest debugging tips or strategies."`
   - Usage: `!debug My JavaScript function isn’t returning the expected result.` - leads to suggestions for debugging the issue.

5. !outline:
   - Definition: `!outline = "Outline the steps of the specified algorithm without coding details."`
   - Usage: `!outline Explain quicksort algorithm.` - results in a step-by-step outline of the quicksort algorithm.

By incorporating these snippets, users could effectively streamline their interactions with Copilot, receiving responses that are more aligned with their immediate needs and preferences.

**Comment by Akash1134** [comment]
Hi @devcatalin ,

Thanks for sharing this feedback. You can check our [public roadmap](https://github.com/github/roadmap/projects/1) to see the kinds of work we are focusing on for the coming year. Smaller fixes won’t be listed, but are often included when work is happening adjacent to it.

We have added relevant labels to this discussion for visibility. As a GitHub user, the most you can do is to share your feedback on this Discussion space with the correct tags (include examples), and trust that we are always working to make GitHub as awesome as possible. We can't guarantee that every bit of feedback sees action, but we are paying attention.

**Comment by devcatalin** [comment]
After further research of the topic, I found that this feature might not be needed since it's possible to use text replacement software directly in the OS.
MacOS has text replacement built in, under the Keyboard settings, and there are multiple software options for Windows.

**Comment by devcatalin** [comment]
As an update, I tried the text replacement setting from macOS and it looks like it doesn't work in the Copilot Chat input :(

I found a [related vscode issue](https://github.com/microsoft/vscode/issues/163134) since the replacements do not work in the code editor either.

Any recommendations on what could be done next?
